docs: adding markup reference docs (#883)
unoplatform/uno.extensions: doc/Reference/Markup/Templates.md

		.Default(() => new TextBlock().Text("Some Vehicle"))
		.Case(v => v.Year < 1960, () => new TextBlock().Text(() => vehicle.Model)))
		.Case<Car>(car => car.Doors > 2, car => new StackPanel()
			.Children(
				new Image().Source(Resource.Static<ImageSource>("Sedan.png")),
				new TextBlock().Text(() => car.Model)))
		.Case<Truck>(truck => new StackPanel().Children(
			new Image().Source(Resource.Static<ImageSource>("truck.png")),
			new TextBlock().Text(() => truck.TowingCapacity)
		))
```

## ContentTemplate

Similar to DataTemplates, ContentTemplates are a special case which must take into consideration the target type of the parent control it is being applied to. This has handled automatically for you by the generated extensions. Similar to the DataTemplate, the method will be strongly typed, only the type will be that of the parent control.

```cs
new Button()
	.Content("Click Me")
	.Template(button => new Border()
		.RenderTransform(new RotateTransform().Angle(45))
		.Child(new ContentPresenter()
			.Content(x => x.TemplateBind(() => button.Content))
```

---
uid: Reference.Markup.Templates
---
## All other Framework Templates

> In all other cases Template properties are treated the same with a simple delegate expected to return the desired UI. In these cases it is assumed that you have the same context as the parent and have no need for additional typing on the delegate.

Reviewer: ```suggestion
In all other cases, `Template` properties are treated the same with a simple delegate expected to return the desired UI. In these cases, it is assumed that the same context as the parent is available and there is no need for additional typing on the delegate.
```
